pkg/kubelet/sysctl/whitelist.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sysctl
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// Whitelist provides a list of allowed sysctls and sysctl patterns (ending in *)
+// and a function to check whether a given sysctl matches this list.
+type Whitelist struct {
+	sysctls  map[string]Namespace
+	prefixes map[string]Namespace
+}
+
+// NewWhitelist creates a new Whitelist from a list of sysctls and sysctl pattern (ending in *).
+func NewWhitelist(sysctls []string) (*Whitelist, error) {
+	w := &Whitelist{
+		sysctls:  map[string]Namespace{},
+		prefixes: map[string]Namespace{},
+	}
+
+	for _, s := range sysctls {
+		if strings.HasSuffix(s, "*") {
```

Comment:
I extended the validation. So we return good error messages. I think that's better than allowing spaces.